src/main/java/de/novatec/domainprimitives/object/ComposedValueObject.java

```diff
@@ -0,0 +1,68 @@
+package de.novatec.domainprimitives.object;
+
+import de.novatec.domainprimitives.validation.InvariantException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.lang.String.format;
+
+public abstract class ComposedValueObject {
+
+    private final List<String> violations;
+
+    protected ComposedValueObject() {
+        this.violations = new ArrayList<>();
+    }
+
+    /**
+     * Providing a starting point this stub could be used check on invariant violations.
+     */
+    protected abstract void validate();
```

Comment:
Imho we should move the structure to be more in line with the PrimitiveType and force the validation within the constructor.

Reply:
Do you have something in mind? 
I couldn't think of a way to run the validation automatically after the variable assignment. Maybe I wasn't creative enough 😄

Reply:
Opened a new Issue für that: #2